backend/src/src-bdpimport/src/main/java/com/bosch/osmi/sw360/bdp/datasink/thrift/ThriftUploader.java

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright (c) Bosch Software Innovations GmbH 2015.
+ * Part of the SW360 Portal Project.
+ *
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License Version 2.0 as published by the
+ * Free Software Foundation with classpath exception.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU General Public License version 2.0 for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program (please see the COPYING file); if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+package com.bosch.osmi.sw360.bdp.datasink.thrift;
+
+import com.bosch.osmi.bdp.access.api.model.ProjectInfo;
+import com.bosch.osmi.sw360.bdp.datasource.BdpApiAccessWrapper;
+import com.bosch.osmi.sw360.bdp.entitytranslation.BdpComponentToSw360ComponentTranslator;
+import com.bosch.osmi.sw360.bdp.entitytranslation.BdpComponentToSw360ReleaseTranslator;
+import com.bosch.osmi.sw360.bdp.entitytranslation.BdpLicenseToSw360LicenseTranslator;
+import com.bosch.osmi.sw360.bdp.entitytranslation.BdpProjectInfoToSw360ProjectTranslator;
+import com.siemens.sw360.datahandler.thrift.RequestStatus;
+import com.siemens.sw360.datahandler.thrift.components.Component;
+import com.siemens.sw360.datahandler.thrift.components.Release;
+import com.siemens.sw360.datahandler.thrift.licenses.License;
+import com.siemens.sw360.datahandler.thrift.projects.Project;
+import com.siemens.sw360.datahandler.thrift.users.User;
+import com.siemens.sw360.datahandler.thrift.bdpimportstatus.BdpImportStatus;
+import org.apache.log4j.Logger;
+import org.apache.thrift.TException;
+
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+// TODO: this contains logic and knowledge of the inner relations => move them to entitytranslation?
+public class ThriftUploader {
```

Comment:
True. But we can also use the git history for the authorship. We would prefer to leave it as it is.